### DISCUSSION TITLE
MarkAsRead не работает

### DISCUSSION THREAD

**Discussion by tebafaz** [SevereCloud/vksdk#212] [Q&A]
Метод messages.MarkAsRead не работает. В целом код выглядит так:

```
package main

import (
	"context"
	"log"
	"os"

	"github.com/SevereCloud/vksdk/v2/api"
	"github.com/SevereCloud/vksdk/v2/events"
	"github.com/SevereCloud/vksdk/v2/longpoll-bot"
	"github.com/joho/godotenv"
)

func init() {
	godotenv.Load()
}

func main() {
	token := os.Getenv("GROUP_TOKEN")
	vk := api.NewVK(token)
	vk.Version = "5.81"
	// get information about the group
	groupParams := api.Params{
		"group_id": os.Getenv("GROUP_ID"),
	}
	group, err := vk.GroupsGetByID(groupParams)
	if err != nil {
		log.Fatal(err)
	}

	// Initializing Long Poll
	lp, err := longpoll.NewLongPoll(vk, group[0].ID)
	if err != nil {
		log.Fatal(err)
	}

	// New message event
	lp.MessageNew(func(_ context.Context, obj events.MessageNewObject) {
		mar := api.Params{
			"peer_id":                   obj.Message.PeerID,
			"mark_conversation_as_read": 1,
			"start_message_id":          obj.Message.ID,
		}
		_, err = vk.MessagesMarkAsRead(mar)
		if err != nil {
			log.Fatal(err)
		}

	})

	// Run Bots Long Poll
	log.Println("Start Long Poll")
	if err := lp.Run(); err != nil {
		log.Fatal(err)
	}
}
```

Ответ приходит {"response":1}, однако в беседе никаких изменении. Также пробовал в start_message_id вместо obj.Message.ID ставить obj.Message.ConversationMessageID, и пытался менять mark_conversation_as_read на 0. Версию тоже менял.

Авторизация проходила успешно, потому что в логах можно было видеть текст сообщения. Пробовал метод с примера отправки сообщения pong на сообщение ping, тоже работает. Пробовал и на веб сайте https://dev.vk.com/method/messages.markAsRead отправить с теми же параметрами. Приходит response: 1, но изменении в чате нет. Попробовал метод messages.pin в веб браузере, для проверки peer_id, который отправляю в messages.MarkAsRead, там он сработал.

Не думаю что баг касется вас, но и не особо понимаю к кому обращаться.

**Comment by SevereCloud** [comment]
Привет, запустил твой код и он работает

<img width="601" alt="image" src="https://user-images.githubusercontent.com/14944123/179308332-366e99f3-d421-4599-ba60-c44ba62698ab.png">
<img width="597" alt="image" src="https://user-images.githubusercontent.com/14944123/179308434-5112b59c-3356-4820-9855-fff61c0f69e7.png">

Попробуй с другой группой. Если в ней получится, то напиши мне ID группы с багом в [лс](https://vk.me/severecloud)

**Reply by tebafaz** [reply]
Оказывается эта ошибка возникает именно в беседе с чат ботом, а в личных сообщениях с пользователем такого не происходит. Он и вправду в таком случае читает

Когда peer_id 2000000001 он не маркирует как прочитанное
